jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditorDialog.kt

```diff
@@ -0,0 +1,151 @@
+        super.doCancelAction()
+    }
+
+    override fun doOKAction() {
+        // Do nothing, close logic is handled separately
+    }
+
+    private fun getRelativeTime(unitOfTime: String, relTimeNumber: Long): StartEndDate {
+
+        val endDate = System.currentTimeMillis() / 1000
+        val startDate: Long = when (unitOfTime) {
+            "Minutes" -> endDate - (relTimeNumber * 60)
+            "Hours" -> endDate - (relTimeNumber * 60 * 60)
+            "Days" -> endDate - (relTimeNumber * 60 * 60 * 24)
+            "Week" -> endDate - (relTimeNumber * 60 * 60 * 24 * 7)
+            else -> 0
+        }
+        return StartEndDate(startDate, endDate)
+    }
+    private fun getAbsoluteTime(startDate: Date, endDate: Date): StartEndDate = StartEndDate(((startDate.time) / 1000), ((endDate.time) / 1000))
+
+    private fun getFilterQuery(searchTerm: String): String = "fields @message, @timestamp | filter @message like '$searchTerm'"
+
+    private fun beginQuerying() {
+        if (!okAction.isEnabled) {
+            return
+        }
+        val funDetails = getFunctionDetails()
+        val queryStartDate: Long
+        val queryEndDate: Long
+        val queryStartEndDate: StartEndDate
+        queryStartEndDate = if (funDetails.absoluteTimeSelected) {
+            getAbsoluteTime(funDetails.qStartDateAbsolute!!, funDetails.qEndDateAbsolute!!)
+        } else {
+            getRelativeTime(funDetails.qRelativeTimeUnit, funDetails.qRelativeTimeNumber.toLong())
+        }
+        queryStartDate = queryStartEndDate.startDate
+        queryEndDate = queryStartEndDate.endDate
+
+        val query = if (funDetails.queryingLogsSelected) {
+            funDetails.qQuery } else {
+            getFilterQuery(funDetails.qSearchTerm)
+        }
+        close(OK_EXIT_CODE)
+        qQueryingLogGroupApiCall.executeStartQuery(queryEndDate, funDetails.logGroupName, query, queryStartDate)
+    }
+    private fun getFunctionDetails(): QueryDetails = QueryDetails(
+        logGroupName = lGroupName,
+        absoluteTimeSelected = view.absoluteTimeRadioButton.isSelected,
+        qStartDateAbsolute = view.qStartDate.date,
+        qEndDateAbsolute = view.qEndDate.date,
+        relativeTimeSelected = view.relativeTimeRadioButton.isSelected,
+        qRelativeTimeUnit = view.RelativeTimeUnit.selectedItem.toString(),
+        qRelativeTimeNumber = view.RelativeTimeNumber.text,
+        searchTermSelected = view.searchTerm.isSelected,
+        qSearchTerm = view.querySearchTerm.text,
+        queryingLogsSelected = view.queryLogGroupsRadioButton.isSelected,
+        qQuery = view.queryBox.text
+    )
+
+    private inner class QueryLogGroupOkAction : OkAction() {
+        init {
+            putValue(Action.NAME, "Apply")
+        }
+        override fun doAction(e: ActionEvent?) {
+            super.doAction(e)
+            if (doValidateAll().isNotEmpty()) return
+            beginQuerying()
+        }
+    }
+}
+
+class QueryEditorValidator {
```

Comment:
feel like this should either be an `object` or a function in a `companion object` for `QueryEditorDialog`